Feature/table view

wwwsurenio opened the pull request:
I added the TableView to the app, and set it up to show the movie titles for now, next step is to setup the custom cell and show all the data from the array.
Or is there another way I should be doing this?
![Simulator Screen Shot - iPhone 14 - 2023-05-12 at 14 12 30](https://github.com/wwwsurenio/imdb/assets/105368039/32947b9c-39f0-45a7-b9e3-08d9f5c275d7)

wwwsurenio committed:
Created a MovieModel struct, added tableView to MovieViewController

wwwsurenio committed:
Setup for the movieTitle to appear in the TableView

aleksandr-lavrinenko commented on imdb/UI/View/MovieViewController.swift:
```diff
@@ -27,73 +22,61 @@ class MovieViewController: UIViewController {
+    }
+    
+    func createMovieArray() {
+        movies.append(MovieModel(
```
Можем пока создать переменную let movies: [MovieModel] = [
MovieModel(...),
MovieModel(..)
]

aleksandr-lavrinenko commented on imdb/UI/View/MovieViewController.swift:
```diff
@@ -27,73 +22,61 @@ class MovieViewController: UIViewController {
         ])
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cellId")
```
Все правильно - теперь попробуй UITableViewCell заменить на MovieCell!

wwwsurenio committed:
Refactor movie array into a constant and update table view cell in MovieViewController
- Refactored `createMovieArray` into a constant array of `MovieModel` objects for better performance and readability.
- Updated `cellForRowAt` method to dequeue a `MovieTableCell` and set its properties to the corresponding `MovieModel`.
- Removed unused `blah` method.
- Updated cell registration to use `MovieTableCell`
Added properties and UI elements for movie thumbnail, title, description and rating in MovieTableCell.
Implemented the layout for the MovieTableCell, including constraints for the thumbnail image view, title label, description label and rating label.

wwwsurenio commented:
**Created a custoom cell - MovieTableCell**
Added properties for each movie in the table view cell.
Added private variables to organize and configure each element within the table view cell.
Created and used the anchor function to set up constraints for the layout of the table view cell's content.
Removed unnecessary comments.

Code refactored based on comments above.

Current results:
![Simulator Screen Shot - iPhone 14 - 2023-05-13 at 21 50 32](https://github.com/wwwsurenio/imdb/assets/105368039/57c99667-a12f-421b-bfba-1fae195e5b15)

wwwsurenio committed:
added more films, to make sure scrolling works

wwwsurenio commented:
Added more films to test scrolling
![Simulator Screen Shot - iPhone 14 - 2023-05-13 at 22 33 14](https://github.com/wwwsurenio/imdb/assets/105368039/347b8554-9735-4f38-8a1a-62ad094abe45)
